test/client_bandit_test.py

```diff
@@ -0,0 +1,159 @@
+# Note: contains tests for client.py related to bandits to avoid
+# making client_test.py too long.
+
+
+import json
+import os
+from time import sleep
+from typing import Dict
+from eppo_client.bandit import BanditResult, ActionContext, Attributes
+
+import httpretty  # type: ignore
+import pytest
+
+from eppo_client.assignment_logger import AssignmentLogger
+from eppo_client.configuration_requestor import BANDIT_ENDPOINT, UFC_ENDPOINT
+from eppo_client import init, get_instance
+from eppo_client.config import Config
+
+TEST_DIR = "test/test-data/ufc/bandit-tests"
+FLAG_CONFIG_FILE = "test/test-data/ufc/bandit-flags-v1.json"
+BANDIT_CONFIG_FILE = "test/test-data/ufc/bandit-models-v1.json"
+test_data = []
+for file_name in [file for file in os.listdir(TEST_DIR)]:
+    with open("{}/{}".format(TEST_DIR, file_name)) as test_case_json:
+        test_case_dict = json.load(test_case_json)
+        test_data.append(test_case_dict)
+
+print(test_data)
+
+MOCK_BASE_URL = "http://localhost:4001/api"
+
+DEFAULT_SUBJECT_ATTRIBUTES = Attributes(
+    numeric_attributes={"age": 30}, categorical_attributes={"country": "UK"}
+)
+
+
+class MockAssignmentLogger(AssignmentLogger):
+    def log_assignment(self, assignment_event: Dict):
+        print(f"Assignment Event: {assignment_event}")
+
+    def log_bandit_action(self, bandit_event: Dict):
+        print(f"Bandit Event: {bandit_event}")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def init_fixture():
+    httpretty.enable()
+    with open(FLAG_CONFIG_FILE) as mock_ufc_response:
+        ufc_json = json.load(mock_ufc_response)
+
+    with open(BANDIT_CONFIG_FILE) as mock_bandit_response:
+        bandit_json = json.load(mock_bandit_response)
+
+    httpretty.register_uri(
+        httpretty.GET,
+        MOCK_BASE_URL + UFC_ENDPOINT,
+        body=json.dumps(ufc_json),
+    )
+    httpretty.register_uri(
+        httpretty.GET,
+        MOCK_BASE_URL + BANDIT_ENDPOINT,
+        body=json.dumps(bandit_json),
+    )
+    client = init(
+        Config(
+            base_url=MOCK_BASE_URL,
+            api_key="dummy",
+            assignment_logger=AssignmentLogger(),
+        )
+    )
+    sleep(0.1)  # wait for initialization
```

Comment:
🤨

Reply:
Otherwise the test for whether the client is initialized is run before initialization